test/json_test.go

```diff
@@ -0,0 +1,83 @@
+package test
+
+import (
+	"github.com/brandonromano/wrecker/test/models"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPostJSON(t *testing.T) {
+
+	response := models.Response{
+		Content: new(models.User),
+	}
+
+	userIn := models.User{
+		Id:       98,
+		UserName: "Steve Rogers",
+		Location: "New York, NY",
+	}
+
+	_, err := wreckerClient.Post("/users").
+		WithBody(userIn).
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	userOut, ok := response.Content.(*models.User)
+
+	assert.True(t, ok)
+	assert.True(t, userOut.Id == 98)
+	assert.True(t, userOut.UserName == "Steve Rogers")
+	assert.True(t, userOut.Location == "New York, NY")
+}
+
+func TestPutJSON(t *testing.T) {
+
+	response := models.Response{
+		Content: new(models.User),
+	}
+
+	userIn := models.User{
+		Id:       99,
+		UserName: "Natasha Romanov",
+		Location: "New York, NY",
+	}
+
+	_, err := wreckerClient.Put("/users").
+		WithBody(userIn).
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	userOut, ok := response.Content.(*models.User)
+
+	assert.True(t, ok)
+	assert.True(t, userOut.Id == 99)
+	assert.True(t, userOut.UserName == "Natasha Romanov")
+	assert.True(t, userOut.Location == "New York, NY")
+}
+
+func TestPutString(t *testing.T) {
+
+	response := models.Response{
+		Content: new(string),
+	}
+
+	_, err := wreckerClient.Put("/status").
+		WithBody("status code green").
+		Into(&response).
+		Execute()
+
+	assert.True(t, err == nil)
+	assert.True(t, response.Success)
+
+	status, ok := response.Content.(*string)
+
+	assert.True(t, ok)
+	assert.True(t, *status == "status code green")
```

Comment:
So the test is passing, but I actually don't see how this is working now that we've removed support for text in the `prepareRequestBody` function... Hmm, perhaps you know why this is passing?